astropy/timeseries/periodograms/lombscargle/tests/test_lombscargle.py

```diff
@@ -97,7 +101,11 @@ def test_all_methods(
         fit_mean=fit_mean,
         normalization=normalization,
     )
-    P_expected = ls.power(frequency)
+    # Reference defaults to "scipy" in most cases
+    if method == "scipy":
+        P_expected = ls.power(frequency, method="cython")
+    else:
+        P_expected = ls.power(frequency)
```

Comment:
This change seems strange to me – the default is `scipy` in some cases, but not in others. If it's important to do this for `scipy`, why is it not important to do this in other cases?

Reply:
I did not have a full overview of the defaults, but it seems for this dataset (`not prefer_fast`) the only other default would be `cython`, so using `slow` in that case would probably make sense. I guess in principle the comparisons should be symmetric, but when debugging the `center_data` settings I was not so sure. Anyway would make more sense than to compare a method to itself imo.

Reply:
Sure, but you're still comparing a method to itself whenever the default does not happen to be `"scipy"` – that's what seems strange here.

Reply:
No, only for the `cython` case I'd add – or are there any configurations I have overlooked when `auto` would not choose either `cython` or `scipy`? Anyway that was the behaviour before, so just leaving it that way does not seem useful.

Reply:
Basically this was trying to take up @mhvk's suggestion to [test the 'scipy' method against one of the internals](https://github.com/astropy/astropy/issues/17149#issuecomment-2406040678), which was not (explicitly) done before. The issues I also have with the existing construct are
1. ~¼ of the tests is not really comparing anything than repeating the same function call.
2. 'scipy', and for some cases 'cython', is only ever tested indirectly (when some other methods are failing against it as the reference), which is creating confusing diagnostics in case that 'scipy' is broken – e.g. in #17149 it initially appeared that all other methods were broken until someone remembered that 'scipy' was used as the baseline for those tests.

Reply:
OK, then if (1) is a problem, I think we should solve it for all test runs, rather than just for some.